cobspec/src/main/java/com/saralein/cobspec/controller/LogController.java

```diff
@@ -0,0 +1,58 @@
+package com.saralein.cobspec.controller;
+
+import com.saralein.cobspec.data.LogStore;
+import com.saralein.server.controller.Controller;
+import com.saralein.server.request.Request;
+import com.saralein.server.response.Response;
+import java.util.Base64;
+
+public class LogController implements Controller {
+    private final LogStore logStore;
+    private final String username;
+    private final String password;
+
+    public LogController(LogStore logStore, String username, String password) {
+        this.username = username;
+        this.password = password;
+        this.logStore = logStore;
+    }
+
+    public Response respond(Request request) {
+        return isAuthorized(request) ? createAuthorizedResponse() : createUnAuthorizedResponse();
+    }
+
+    private Response createAuthorizedResponse() {
+        return new Response.Builder()
+                .status(200)
+                .addHeader("Content-Type", "text/plain")
+                .body(logStore.retrieveLog())
+                .build();
+    }
+
+    private Response createUnAuthorizedResponse() {
+        return new Response.Builder()
+                .status(401)
+                .addHeader("WWW-Authenticate", "Basic realm=\"ServerCity\"")
+                .build();
+    }
+
+    private String parseRequestAuthorization(Request request) {
+        String encodedAuthorization = request.getHeader("Authorization");
+
+        if (encodedAuthorization.isEmpty()) {
+            return null;
+        }
+
+        return encodedAuthorization.split(" ")[1];
+    }
+
+    private String encodeValidAuthorization() {
+        String validAuthorization = username + ":" + password;
+        return Base64.getEncoder().encodeToString(validAuthorization.getBytes());
+    }
```

Comment:
Encoding and parsing request/response bits feel like they don't belong as a responsibility of a controller

Reply:
(maybe there are a class (or two?) that can be extracted)

Reply:
Broke this up.